### ISSUE TITLE
 Zimit 2 tests: mesquartierschinois

### ISSUE THREAD

**Issue by benoit74** [openzim/zim-requests#1014]
A new Zimit2 ZIM is ready for testing

Scraper: warc2zim 2.0.0-dev9 + zimit 2.0.0-dev5 + Browsertrix crawler 1.1.3
ZIM: https://mirror.download.kiwix.org/zim/.hidden/dev/mes-quartiers-chinois_fr_all_2024-05.zim
Library link: https://dev.library.kiwix.org/viewer#mes-quartiers-chinois_fr_all_2024-05

Suggested test plan:

- Open the ZIM, check homepage and 5 subpages (randomly chosen) loads properly
- Perform a suggestion search and open any suggestion
- Perform a full-text search, check result quality and open any result
- Confirm Youtube video plays correctly (audio + video), see e.g. Girona article

Known issues:
- only first seconds of Vimeo videos are working (this is not planned to be fixed soon)

@Jaifroid, still welcomed (I sent you an invite to this repo with Read rights so that we can at least tag you correctly, if you don't mind of course)

**Comment by Popolechien** [comment]
Kiwix-server: infinite scroll does not work and new entries do not load. Link on the right (categories) return an error but this I suspect is more a dev.library.kiwix.org issue as it is intermittent.
macOS: file does not open at all
Kiwix-JS (Chrome): video not showing (still black screen), rest looks great.
Android: file isn't picked up by Kiwix

**Comment by Jaifroid** [comment]
> @Jaifroid, still welcomed (I sent you an invite to this repo with Read rights so that we can at least tag you correctly, if you don't mind of course)

Accepted, thanks.

**Comment by Jaifroid** [comment]
In PWA and Browser Extension (installed via Chromium browser) the video is working for me on Windows. Stéphane, for the Borwser Extension, please check it says v4.0.1 top left?

![image](https://github.com/openzim/zim-requests/assets/4304337/ad6dfa33-365b-4ee8-a4bf-09a610a89a4c)

**Comment by Popolechien** [comment]
Ah no it's the 4.0.0

How come this does not update automatically? I feel cheated!
Let me look into this and revert back.

**Comment by Jaifroid** [comment]
@Popolechien No, not your fault. Please check in Configuration that you're using ServiceWorker mode, not ServiceWorkerLocal mode. To run a lot of inline JS in the browser, the app needs to be in ServiceWorker mode, which will access the server to cache the latest code. That should then show v4.0.1.

**Comment by Popolechien** [comment]
Yup we're good. But then now I also get the endless wheel of loading at the bottom of the page:

![Screenshot 2024-05-30 at 17 11 02](https://github.com/openzim/zim-requests/assets/21675885/2bd46f71-80f5-42ae-8c69-3f47cc1c2d45)

**Comment by Jaifroid** [comment]
Is that on the Home / landing page? I don't get it there, but I do get it on other pages. I think it's a scraper issue.

**Comment by Popolechien** [comment]
Home page, but also on individual posts. I think it's a scraper issue yes because on [one of the pages](https://dev.library.kiwix.org/viewer#mes-quartiers-chinois_fr_all_2024-05/mesquartierschinois.wordpress.com/2013/10/08/un-avec-la-vague/) I can see the `Chargement` (loading) being hardcoded text

**Comment by Popolechien** [comment]
Ah found another issue when clicking on Related posts at the bottom. Updated my earlier comment.

**Comment by Jaifroid** [comment]
@Popolechien I see what you mean about infinite scrolling on Landing Page, sorry -- it gets a long way, as far as "Supersize Me", and then stops. I suspect the crawler stopped there. When I said infinite scrolling was working, it was because I hadn't scrolled so far down!

**Comment by Jaifroid** [comment]
> Ah found another issue when clicking on Related posts at the bottom. Updated my earlier comment.

I confirm "Articles similaires" are not scraped, and appear as external links. This may be a domain scope issue.

**Comment by Jaifroid** [comment]
Finally got a console readout for iPhone 15 Pro Max with iOS 17 (text below image). Note that in this format (iPhone as opposed to iPad) the Vimeo video is also not found. I suspect a specific iPhone version is being requested by the player, and this is not scraped. This is going to be a significant issue I think, unless we can find a fallback mechanism to forcibly inject the MP4 that was scraped into the video player (this is something that @mgautierfr tried with some success previously).

![image](https://github.com/openzim/zim-requests/assets/4304337/baa23c54-7596-441a-bb34-d0bf11c52fab)

```
urlRewriten:
	- current_url: https://pwa.kiwix.org/mes-quartiers-chinois_fr_all_2024-05.zim/C/youtube.fuzzy.replayweb.page/embed/1T3u9jjXcGM
	- orig_host: www.youtube.com
	- orig_scheme: https
	- orig_url: https://www.youtube.com/embed/1T3u9jjXcGM?version=3&rel=1&showsearch=0&showinfo=1&iv_load_policy=1&fs=1&hl=fr&autohide=2&wmode=transparent
	- prefix: https://pwa.kiwix.org/mes-quartiers-chinois_fr_all_2024-05.zim/C/
	- url: https://i.ytimg.com/vi/1T3u9jjXcGM/sddefault.jpg?sqp=-oaymwEmCIAFEOAD8quKqQMa8AEB-AG-B4AC0AWKAgwIABABGHIgVig-MA8=&rs=AOn4CLBoBLbDfVnzoGC5qO5p5z-AKuvyIg
	- useRel: false
	- mod: im_
	- doc: undefined
	- finalUrl: https://pwa.kiwix.org/mes-quartiers-chinois_fr_all_2024-05.zim/C/i.ytimg.com.fuzzy.replayweb.page/vi/1T3u9jjXcGM/thumbnail.jpg
	
VM20 wombatSetup.js:3 urlRewriten:
	- current_url: https://pwa.kiwix.org/mes-quartiers-chinois_fr_all_2024-05.zim/C/youtube.fuzzy.replayweb.page/embed/1T3u9jjXcGM
	- orig_host: www.youtube.com
	- orig_scheme: https
	- orig_url: https://www.youtube.com/embed/1T3u9jjXcGM?version=3&rel=1&showsearch=0&showinfo=1&iv_load_policy=1&fs=1&hl=fr&autohide=2&wmode=transparent
	- prefix: https://pwa.kiwix.org/mes-quartiers-chinois_fr_all_2024-05.zim/C/
	- url: https://i.ytimg.com/vi/1T3u9jjXcGM/sddefault.jpg?sqp=-oaymwEmCIAFEOAD8quKqQMa8AEB-AG-B4AC0AWKAgwIABABGHIgVig-MA8=&rs=AOn4CLBoBLbDfVnzoGC5qO5p5z-AKuvyIg
	- useRel: undefined
	- mod: undefined
	- doc: undefined
	- finalUrl: https://pwa.kiwix.org/mes-quartiers-chinois_fr_all_2024-05.zim/C/i.ytimg.com.fuzzy.replayweb.page/vi/1T3u9jjXcGM/thumbnail.jpg
	
bundle.min.js:24 Title C/www.youtube.com/api/stats/qoe?fmt=18&cpn=yqsCyjsNQbmf6n6y&el=embedded&ns=yt&fexp=v1,24004644,76094,54572,304051,104308,16747,6271,26443548,7111,36343,9954,119... not found in archive.
bundle.min.js:24 Title C/i.ytimg.com.fuzzy.replayweb.page/vi/uapAbBOcy6U/thumbnail.jpg... not found in archive.
```

**Comment by benoit74** [comment]
Thank you!

> infinite scroll not working

This is not a zimit2 issue, this is an bad recipe configuration. Autoscroll is needed on this website, never been configured, never worked.

> Link on the right (categories) return an error but this I suspect is more a dev.library.kiwix.org issue as it is intermittent.

I can't repro this, is anyone else impacted?

> macOS: file does not open at all Unable to load the article requested.

Issue is probably same as fas-military-medicine, I commented in this one at least: https://github.com/kiwix/kiwix-apple/issues/779

> Safari Youtube video

Errors you get are not related AFAIK, all missing thumbnails are just thumbnails (i.e. not the video itself) and come from the preview of linked videos that Youtube player would like to show, crawler did not fetched them. Maybe we should add a generic custom CSS to hide all these preview which are not available anyway. And the error on /api/stats looks a lot like a tracker (which obviously is not in the ZIM ...).

I reproduce the error on pwa.kiwix.org on Safari on MacOS, so this is an issue of PWA on safari as far as I can tell. Especially since same video in same ZIM in same Safari on same machine works well. And it also works on PWA on Firefox on same machine. No reason for the JS player to choose a different URL only because it is running inside a PWA on Safari. Probably many reason for the PWA to do something silly on Safari due to all Safari subtleties.

I also reproduced the error with other ZIMs, so not specific to mesquartierschinois.

I do not trust the argument about specific video format for specific reader anymore, did not saw it, so far problems on Youtube videos was always related to ZIM content (improper URL rewriting or Wombat injection). And again, does not work on PWA while it works on kiwix-serve.

I opened https://github.com/kiwix/kiwix-js/issues/1256

> Clicking on "related posts" at the bottom of an individual entry returns an error `Not found`

This is an interesting one... opened https://github.com/openzim/warc2zim/issues/291

**Comment by benoit74** [comment]
Nota: shouldn't we get rid via custom CSS of these two sections:

![image](https://github.com/openzim/zim-requests/assets/7102089/85eef6b8-c94a-45b4-b227-ba35adb39a21)

![image](https://github.com/openzim/zim-requests/assets/7102089/3eb5e92b-ceb0-4fa4-b29f-137245dee014)

Both are mostly meaningless in an offline context

**Comment by benoit74** [comment]
> Autoscroll is needed on this website, never been configured, never worked.

Recipe updated, not relaunched since we have too many other things to solve first

**Comment by Jaifroid** [comment]
Just to confirm that the Girona video works when served from Kiwix Serve on iOS and macOS Safari (tested also on iPhone and iPad via BrowserStack). So the problems with the PWA and Browser Extension playing Zimit2 YouTube video in Safari (only!) is indeed a weird issue on my side. ☹️

**Comment by Popolechien** [comment]
> shouldn't we get rid via custom CSS of these two sections:

Not for this specific website as this is only a test one. If however we can make it a regular feature for all other wordpress blogs out there, then yes.

**Comment by Popolechien** [comment]
I understand this issue is for testing purposes, but should it really be kept open, seeing as it's not an active request for the library?

**Comment by benoit74** [comment]
@Jaifroid do you have anything left to test / fix in PWA and Browser Extension (can you please open an issue with details in relevant repo then) or are we good to close this issue?

**Comment by Jaifroid** [comment]
@benoit74 Apologies for slow reply, only just saw this (I get so many GitHub notifications it's hard to pick out the important ones!). Please do close. Any specific bugs / browser incompatibilities on my side are tracked separately indeed.
